File: heronpy/dsl/impl/contextimpl.py
# Copyright 2016 Twitter. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""module for contextimpl: ContextImpl"""

from heronpy.dsl.context import Context

class ContextImpl(Context):
  """ContextImpl"""
  def __init__(self, topology_context, state):
    self._topology_context = topology_context
    self._state = state

  def get_task_id(self):
    return self._topology_context.get_task_id()

  def get_config(self):
    return self._topology_context.get_cluster_config()

  def get_stream_name(self):
    return self._topology_context.get_this_sources().keys()[0].id

  def get_state(self):
    return self._state

File: heron/api/src/java/com/twitter/heron/dsl/impl/ContextImpl.java
//  Copyright 2017 Twitter. All rights reserved.
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

package com.twitter.heron.dsl.impl;

import java.io.Serializable;
import java.util.Map;

import com.twitter.heron.api.state.State;
import com.twitter.heron.api.topology.TopologyContext;
import com.twitter.heron.dsl.Context;

/**
 * Context is the information available at runtime for operators like transform.
 * It contains basic things like config, runtime information like task,
 * the stream that it is operating on, ProcessState, etc.
 */
public class ContextImpl implements Context {
  private TopologyContext topologyContext;
  private Map<String, Object> topologyConfig;
  private State<Serializable, Serializable> state;

  public ContextImpl(TopologyContext topologyContext, Map<String, Object> topologyConfig,
                     State<Serializable, Serializable> state) {
    this.topologyContext = topologyContext;
    this.topologyConfig = topologyConfig;
    this.state = state;
  }

  @Override
  public int getTaskId() {
    return topologyContext.getThisTaskId();
  }

  @Override
  public Map<String, Object> getConfig() {
    return topologyConfig;
  }

  @Override
  public String getStreamName() {
    return topologyContext.getThisStreams().iterator().next();
  }

  @Override
  public int getStreamPartition() {
    return topologyContext.getThisTaskIndex();
  }

  @Override
  public State<Serializable, Serializable> getState() {
    return state;
  }
}
